app/models/early_career_teacher_profile.rb

```diff
@@ -4,6 +4,9 @@ class EarlyCareerTeacherProfile < ApplicationRecord
   belongs_to :user
   belongs_to :core_induction_programme, optional: true
   belongs_to :cohort, optional: true
+  belongs_to :mentor_profile, optional: true
+
+  has_one  :mentor, through: :mentor_profile, source: :user
```

Comment:
I would group these two lines together - logical connection trumps semantics and syntax.